File: src (cpp)/infrastructure/facility/restaurant.cpp
#include "restaurant.h"

Restaurant::Restaurant(bool accessible, const std::string& _name): Facility(accessible), def_facility_type("Restaurant"), name(_name) {
  facility_type = def_facility_type;
}
Restaurant::Restaurant(const Restaurant& R): Facility(R.is_accessible), def_facility_type("Restaurant"), name(R.name) {
  facility_type = R.facility_type;
}
Restaurant& Restaurant::operator=(const Restaurant& R) {
  name = R.name;
  facility_type = R.facility_type;
  is_accessible = R.is_accessible;
  return *this;
}
std::string Restaurant::GetRestaurantName() {
  return name;
}

char Restaurant::Render() {
  return 'R';
}

File: src/virtualzoo/infrastructure/facility/Restaurant.java
package virtualzoo.infrastructure.facility;

/**
 * Kelas Restaurant mendefinisikan sebuah tempat makan di dalam Zoo.
 *
 * @author Oktavianus Handika - 13515035
 * @version 3.0
 * @since 3.0
 */
public class Restaurant extends Facility {

  /**
   * Nama untuk setiap Restaurant.
   */
  protected String name;

  /**
   * Constructor
   *
   * Membuat sebuah restaurant.
   */
  public Restaurant(boolean accessible, String facilityName) {
    super(accessible);
    facilityType = "Restaurant";
    name = facilityName;
  }

  /**
   * Mengembalikan suatu char yang menandakan Park dalam satu Cell.
   */
  public char render() {
    return 'R';
  }
}